R/simplify_fn.R

```diff
@@ -0,0 +1,36 @@
+simple_simplify_fn <- function(object, predictions) {
+  predictions %>%
+    map(~ .x %>%
+          map(~.x[[1]]) %>%
+          flatten() %>% 
+          as_tibble()) %>%
+    bind_rows()
+}
+
+validate_prediction_simplify_fn <- function(prediction_simplify_fn) {
+  if (length(formals(prediction_simplify_fn)) == 1) {
+    TRUE
+  } else {
+    stop("simplify must be a function with only one argument.")
+  }
+}
+
+simplify_predictions <- function(object, predictions, simplify) {
+  if (is.function(simplify) && validate_prediction_simplify_fn(simplify)) {
+    simplify(predictions)
+  } else {
+    if (isTRUE(simplify)) {
```

Comment:
Again, `TRUE` should mean "auto", which means we'll just use whatever simplifier is registered for custom estimators or bypass it entirely if there is none. 

I think we should keep the argument as `TRUE` rather than changing it to "auto" as I think the logical is simpler and more readily understood. `simplify = TRUE` really means "simplify if possible" .